### ISSUE TITLE
linter: `eslint/no-undef` fails on ts type `NodeListOf`

### ISSUE THREAD

**Issue by Sysix** [oxc-project/oxc#7008]
### What version of Oxlint are you using?

0.10.3

### What command did you run?

oxlint -c .oxlint.json

### What does your `.oxlint.json` config file look like?

```
{
  "env": {
    "browser": true
  },
  "categories": {
    "correctness": "off"
  },
  "rules": {
    "no-undef": "error"
  }
}

```

### What happened?

It does not know about this specific types:

```
  × eslint(no-undef): 'NodeListOf' is not defined.
    ╭─[src/index.ts:11:12]
 10 │ 
 11 │ const xyz: NodeListOf<HTMLElement> = document.querySelectorAll('.xxxx');
    ·            ──────────
 12 │ 
    ╰────

```

Others lines like: `const abc: Element | null = document.querySelector('.xxx');` are working

**Comment by Boshen** [comment]
no-undef is still under nurser, PRs welcome!

The fix should be checking for symbol flags.

**Comment by shulaoda** [comment]
Actually, this will also report an error, and I think this is expected behavior? See [typescript-eslint](https://typescript-eslint.io/play/#ts=5.4.5&fileType=.tsx&code=MYewdgzgLgBAHgTwF4C4YDkQBMCmAZAS2gHkAzAHgAkAVAWTwFEAbHAWxzCgD4YBeGLCGABXdpwB0AR2E4ATggDKOFsCghZAQSZMAFAHJxcI3D0BKANwBYAFBA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQHYHsBaWXAE0QDN0wBtcbHRaafaSeiAXXoF8QegA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)

```bash
  ⚠ eslint(no-undef): 'Element' is not defined.
   ╭─[no_undef.tsx:1:12]
 1 │ const abc: Element | null = document.querySelector('.xxx');
   ·            ───────
   ╰────

  ⚠ eslint(no-undef): 'document' is not defined.
   ╭─[no_undef.tsx:1:29]
 1 │ const abc: Element | null = document.querySelector('.xxx');
   ·                             ────────
   ╰────
```

**Comment by mimbrown** [comment]
I think the issue here is that `NodeListOf` is a typescript-only type, it's not a runtime global. Oxlint also warns on standard Typescript helper types like `Partial`:

```
  ⚠ eslint(no-undef): 'Partial' is not defined.
   ╭─[test.ts:5:10]
 4 │ 
 5 │ const a: Partial<Name> = {}
   ·          ───────
   ╰────
```

So I think the options are:
1. Add support for TypeScript globals, not just JS ones
2. Skip TS types completely for this lint rule
